### PR TITLE
Cmake blas

### PR DESCRIPTION
**Pull request by edisongustavo**
This is a re-creation of https://github.com/apache/incubator-mxnet/pull/14028. I've accidentally deleted my fork of `incubator-mxnet`.

## Description ##

Eases the pain of linking with OpenBLAS using cmake.

This PR adds support to use the provided `OpenBLASConfig.cmake`.

### Justification for the change

The standard way of linking with dependencies in CMake is via the `find_package()` mechanism ([docs](https://cmake.org/cmake/help/latest/command/find_package.html)). This mechanism provides 2 ways of finding the dependencies:
- Config mode (uses a file named `OpenBLASConfig.cmake` (or `openblas-config.cmake`), which are provided by the dependency itself)
- Module mode (uses a file named `FindOpenBLAS.cmake`, which is provided by either cmake or the project compiling the dependency, mxnet in this case)

The preferred way to find a dependency is if they provide their "Config" file, since the authors of the dependency are the ones who know best about the structure of their code and how they should be linked.

When I tried to compile MXNet on Windows it didn't work well. My setup was:

- Visual studio 2017.
- OpenBLAS package from the SourceForge binaries package, as specified in the [documentation to build MXNet on Windows from source](http://mxnet.incubator.apache.org/versions/master/install/windows_setup.html#build-from-source).

It failed because this binary package is compiled with a Visual Studio version prior to 2015. So I would get the error `unresolved external symbol __imp____iob_func`. The reasons are explained here: https://stackoverflow.com/questions/30412951/unresolved-external-symbol-imp-fprintf-and-imp-iob-func-sdl2

Then I tried to use the [conda-forge package](https://anaconda.org/conda-forge/openblas). This almost worked.

On further inspection of the [FindOpenBLAS.cmake](https://github.com/apache/incubator-mxnet/blob/master/cmake/Modules/FindOpenBLAS.cmake) and the conda-forge package, I noticed that MXNet didn't use the provided "cmake Config mode" files (`OpenBLASConfig.cmake`). It tried to find the library all by itself, which is an "anti-pattern" when using cmake. So I thought that this could be an opportunity to be able to contribute to the project by improving this module.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###

- [ ] Add , tests, (and when applicable, API doc)

I have tested this by compiling it on Windows and Linux.

To provide OpenBLAS I have tried:

On Linux:
- compiled OpenBLAS from source with cmake and installed it
- compiled OpenBLAS from source with make and installed it

On Windows:
- used the compiled OpenBLAS package from conda-forge: https://anaconda.org/conda-forge/openblas

## Comments ##

### CODE REVIEW

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

cc @KellenSunderland @larroy for review

**Comment by yinghu5** [comment]
in the PR #14877, we only change the MKLDNN and MKL part, and keep the OpenBLAS untouched.

About #14028,  agree the latest solution, I will update the install guide of windows

If you don't have the Intel Math Kernel Library (MKL) installed, download and install OpenBLAS. Note that you should also download ```mingw64.dll.zip`` along with openBLAS and add them to PATH.
make sure the OpenBL to point to use version 0.3.6 instead.  thanks

**Comment by aaronmarkham** [comment]
@edisongustavo I think the suggested docs changes from your earlier PR will be fine. Please add those changes and that will also trigger CI. However, it looks like your last commit is pretty old, so you probably should rebase first.

**Comment by edisongustavo** [comment]
@aaronmarkham Ok, I will do that!

**Comment by pinaraws** [comment]
@mxnet-label-bot add[Build, pr-awaiting-response]

**Comment by pinaraws** [comment]
@mxnet-label-bot remove[pr-awaiting-review]

File: cmake/Modules/FindOpenBLAS.cmake [hunk 1]
```diff
@@ -14,78 +14,181 @@
...
+      "$ENV{CROSS_ROOT}"
+      "${CROSS_ROOT}"
+      )
+endif()
+
+set(OpenBLAS_LIB_SEARCH_PATHS
+    ${OpenBLAS_LIB_SEARCH_PATHS}
+
+    "$ENV{OpenBLAS_HOME}"
+    "${OpenBLAS_HOME}"
```
Comment (larroy) [inline cmake/Modules/FindOpenBLAS.cmake]: We don't need Openblas_home/include?  Where is this list coming from?

**Comment by piyushghai** [comment]
@edisongustavo Can you address the review comments by @larroy ? 

Also can you rebase with the latest master branch ?

**Comment by larroy** [comment]
@edisongustavo could you please look at the CI failures and the linked github issue?  Thank you!

**Comment by karan6181** [comment]
@edisongustavo Could you please re-trigger the CI?

**Comment by piyushghai** [comment]
@edisongustavo Gentle ping ...

**Comment by edisongustavo** [comment]
Hello everyone, I don't think I will be able to work on this anymore.

Given that Conan support has been merged in https://github.com/apache/incubator-mxnet/pull/13400, then I think this PR is obsolete.

Re-open if you think I'm wrong, but this is just dragging forever.

**Pull request closed without merging**